File: branches/Medusa/Source/Fantasy.Jobs/Management/WCFHostService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.ServiceModel;
using System.Configuration;
using System.Threading;
using Fantasy.ServiceModel;

namespace Fantasy.Jobs.Management
{
    public class WCFHostService : AbstractService
    {
        private ServiceHost[] _serviceHosts;


        public override void InitializeService()
        {
            Thread thread = ThreadFactory.CreateThread(() =>
            {
                Type[] types = (Type[])AddIn.GetTypes("jobService/wcf/serviceHost");
                this._serviceHosts = new ServiceHost[types.Length];
                for (int i = 0; i < types.Length; i++)
                {
                    this._serviceHosts[i] = new ServiceHost(types[i]);
                    this._serviceHosts[i].Open();
                }
            }).WithStart();
           
           
            base.InitializeService();
        }

        public override void UninitializeService()
        {
            //foreach (ServiceHost host in this._serviceHosts)
            //{
            //    host.Close();
            //}
            base.UninitializeService();
        }
    }
}

File: branches/Stheno/Fantasy.Jobs/src/fantasy/jobs/management/WCFHostService.java
package fantasy.jobs.management;

import Fantasy.ServiceModel.*;

public class WCFHostService extends AbstractService
{
	private ServiceHost[] _serviceHosts;


	@Override
	public void InitializeService()
	{
//C# TO JAVA CONVERTER TODO TASK: Lambda expressions and anonymous methods are not converted by C# to Java Converter:
		Thread thread = ThreadFactory.CreateThread(() =>
		{
			java.lang.Class[] types = (java.lang.Class[])AddIn.GetTypes("jobService/wcf/serviceHost");
			this._serviceHosts = new ServiceHost[types.length];
			for (int i = 0; i < types.length; i++)
			{
				this._serviceHosts[i] = new ServiceHost(types[i]);
				this._serviceHosts[i].Open();
			}
		}
	   ).WithStart();


		super.InitializeService();
	}

	@Override
	public void UninitializeService()
	{
		//foreach (ServiceHost host in this._serviceHosts)
		//{
		//    host.Close();
		//}
		super.UninitializeService();
	}
}